ext/crates/fp/src/limb.rs

```diff
@@ -0,0 +1,221 @@
+use std::ops::Range;
+
+pub(crate) use crate::constants::Limb;
+
+use crate::{
+    constants::{BITMASKS, BITS_PER_LIMB, BIT_LENGTHS, ENTRIES_PER_LIMB, PRIME_TO_INDEX_MAP},
+    prime::ValidPrime,
+};
+
+/// A struct containing the information required to access a specific entry in an array of `Limb`s.
+#[derive(Copy, Clone)]
+pub(crate) struct LimbBitIndexPair {
+    pub(crate) limb: usize,
+    pub(crate) bit_index: usize,
+}
+
+/// Return the number of bits an element of $\mathbb{F}_P$ occupies in a limb.
+pub(crate) const fn bit_length<const P: u32>() -> usize {
+    BIT_LENGTHS[PRIME_TO_INDEX_MAP[P as usize]]
+}
+
+/// TODO: Would it be simpler to just compute this at "runtime"? It's going to be inlined anyway.
```

Comment:
I agree that this would probably be an improvement.

Reply:
So I pushed those changes. I waited a while before pushing because I wanted to benchmark thoroughly, but it looks like my setup doesn't want to give me consistent measurements. In any case I don't detect any systematic drop in performance.

One benchmark would seem to indicate that computing the values at runtime actually improves performance at odd primes. If that's really the case, my guess would be that's because LLVM is able to perform a few more optimizations after inlining the functions, because the values are not black boxes anymore.